### ISSUE TITLE
Fury: Provide async/await API

### ISSUE THREAD

**Issue by kylef** [apiaryio/api-elements.js#37]
We should update `fury` to have an async (promise returning) API. This has been started in Protagonist and will depend on the release of Drafter 4.

```js
const { Fury } = require('fury');
const fury = new Fury();
const apiBlueprintParser = require('fury-adapter-apib-parser');
fury.use(apiBlueprintParser);
const openAPI2Parser = require('fury-adapter-swagger');
fury.use(openAPI2Parser);

const source = `
FORMAT: 1A

# My API
## GET /message
+ Response 200 (text/plain)

        Hello World!
`;

const parseResult = await fury.parse({source});
```

- [x] Support async/await in Fury core for validate (`packages/fury`)
- [x] Can use "callbackify"/"promisify" to simplify the process as first step. The logic can be contained within Fury to wrap existing adapters methods in `promisify`
- [x] Simplify some tests to use async/await
- [x] Update all examples in each package README to use `async/await` syntax.
- [x] Add CHANGELOG entry into `packages/fury` in "Enhancements"

**Comment by char0n** [comment]
The only disadvantage here is that we'll not be able to distinguish expected failures and unexpected errors. But with promises, basically every API suffers from this problems and we got used to it.

**Comment by opichals** [comment]
> we'll not be able to distinguish expected failures and unexpected errors. But with promises, basically every API suffers from this problems and we got used to it.

@kylef Perhaps we could introduce a convention for the _callback errors_ to be thrown wrapped into a special Error class to be able to distinguish from others?

**Comment by char0n** [comment]
@opichals there is now standard [error.code](https://nodejs.org/api/errors.html#errors_error_code) property which you can use. The recommended approach is to create new Error type and set  the `code` property internally. Then instead of inspecting the `type` with `instanceof`  of the Error, inspecting it's shape by using `code` property is recommended.

**Comment by kylef** [comment]
Available in Fury 3.0.0-beta.11 (https://github.com/apiaryio/api-elements.js/releases/tag/fury%403.0.0-beta.11)
